pcmanfm/bulkrename.cpp

```diff
@@ -68,17 +69,26 @@ BulkRenamer::BulkRenamer(const Fm::FileInfoList& files, QWidget* parent) {
         }
         baseName.insert(end, QLatin1Char('#'));
     }
-    uint failed = 0;
+    QProgressDialog progress(QObject::tr("Renaming files..."), QObject::tr("Abort"), 0, files.size(), parent);
+    progress.setWindowModality(Qt::WindowModal);
+    int i = 0, failed = 0;
     for(auto& file: files) {
+        progress.setValue(i);
+        if(progress.wasCanceled()) {
+            progress.close();
+            QMessageBox::warning(parent, QObject::tr("Warning"), QObject::tr("Renaming is aborted."));
+            return;
+        }
         auto fileName = QString::fromStdString(file->name());
         QString newName = baseName;
-        newName.replace(QLatin1Char('#'), QString::number(start));
+        newName.replace(QLatin1Char('#'), QString::number(start + i));
         if (newName == fileName || !Fm::changeFileName(file->path(), newName, nullptr, false)) {
             ++failed;
         }
-        ++start;
+        ++i;
     }
-    if(failed == files.size()) {
+    progress.setValue(i);
```

Comment:
Shouldn't this be in the above loop?

Reply:
No it shouldn't.

Reply:
Line 76 has the same call.

Reply:
> Line 76 has the same call.

Not the same. Line 90 is after the loop, when `i` is `files.size()`, which mean 100%.

Reply:
I thought he was asking why a call that should be made every time the loop executes is being made at the end. It makes sense why you did it.